openshift-environment-driver/src/main/java/org/jboss/pnc/environment/openshift/OpenshiftStartedEnvironment.java

```diff
@@ -222,7 +229,24 @@ public OpenshiftStartedEnvironment(
         MDCUtils.getProcessContext().ifPresent(v -> environmetVariables.put("logProcessContext", v));
         environmetVariables.put("resourcesMemory", builderPodMemory(environmentConfiguration, parameters));
 
-        createEnvironment();
+        createEnvironmentWithRetries();
+    }
+
+    /**
+     * call createEnvironment with retries in case an error happens when talking with Openshift server An example is the
+     * dreaded:
+     *
+     * com.openshift.restclient.OpenShiftException: Exception trying to GET https://<openshift
+     * server></openshift>/apis/servicecatalog.k8s.io/v1beta1 response code: 503
+     *
+     * The max number of retries is `creationPodRetry` with an exponential backoff to a maximum of
+     * MAX_CREATION_ENVIRONMENT_RETRY_SECONDS
+     *
+     */
+    private void createEnvironmentWithRetries() {
+        RetryPolicy<Object> retryPolicy = new RetryPolicy<>().withMaxRetries(creationPodRetry)
+                .withBackoff(1, MAX_CREATION_ENVIRONMENT_RETRY_SECONDS, ChronoUnit.SECONDS);
+        Failsafe.with(retryPolicy).run(this::createEnvironment);
```

Comment:
Is this a non-blocking back-off ?

Reply:
Good observation! No it is blocking, but I am running the creation of the object inside a completable future. I was conflicted on how to make it less blocking. If you have any suggestions I'm all ears :)

Reply:
Maybe https://jodah.net/failsafe/async-api/ with our MDC aware MDCScheduledThreadPoolExecutor ?

Reply:
Yeah good suggestion! I think that'd work! Then on `monitorInitialization` we'll chain for the completablefuture for `createEnvironment` together with the other existing checks. That's a bigger change, but let's see what happens :)